### PR TITLE
Add automation for deploying Azure conformance test resources

### PR DESCRIPTION
**Pull request by CodeMonkeyLeet**
# Description

- Add .bicep definitions for Azure deployment of conformance test resources.
- Add setup-azure-conf-test.sh script for deploying .bicep definition and configuring other assets needed for the Azure conformance tests:
   - Creates the appropriate service principals and their credentials for the test.
   - Populates the key vault with the test parameters and secrets used by the GitHub conformance.yml test workflow.
   - Generates a .rc file that can be invoked with `source` to configure the local environment variables and secrets for running conformance tests described in tests/conformance/README.md.
   - Generates a teardown script that can be used to dispose of the deployed Azure resources when done.

## Issue reference

Fixes: #1000

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Extended the documentation

### CODE REVIEW

**Review by artursouza — CHANGES_REQUESTED** [review]
I have a concern about this script not being exercised in the conformance test workflow because it will quickly become stale as people add more test resources manually. Can the conformance test actually use this and then delete the resources in the end?

**Comment by CodeMonkeyLeet** [comment]
> I have a concern about this script not being exercised in the conformance test workflow because it will quickly become stale as people add more test resources manually. Can the conformance test actually use this and then delete the resources in the end?

@artursouza I looked into this when I was developing the script and ultimately scoped it to just the script, focused on easing manual testing; those parts of the script won't be tested through workflow today as is (i.e. secrets are sourced from different locations). There are also a few additional issues with just running the script in the workflow for setup and teardown:

- Setting up and tearing down all the resources bumps up the time by about another 5-10 minutes (considering the cron job runs every 30 minutes, that's not trivial). 
- It also introduces many potential failure points unrelated to the conformance test; in my experience testing it, I've encountered at least 2 non-repeating failures: 
   - a deployment "failure" where a resource did not meet the goal state in time
   - a script failure where one az cli endpoint to obtain info used to populate the test secrets was returning the 500 internal error briefly.
- Because of this, if intent is to always run this in the conformance test workflow, I would propose refactoring the script into a separate set of smaller scripts to be run as GitHub actions, as a follow-up to this PR. i.e. it should skip the manual-testing specific actions such as creating the config .rc file, and cede control flow to GitHub actions so that it always runs cleanup on partial failure etc. (the latter is particularly important because key vaults must be purged before they can be recreated.)
- An alternative is to _add_ a workflow specifically to test the script, which also separates concerns since it doesn't have to run the non-Azure components test matrix. I'd be happy to add that as a separate PR, but I don't have the ability to implement and test the solution end-to-end: It would require creating a [service principal](https://docs.microsoft.com/en-us/cli/azure/create-an-azure-service-principal-azure-cli) under the dapr _production_ subscription and granting it fairly significant privileges to create and purge resources (purging key vaults in particular is a privilege we should be careful with), then enabling those credentials as a GitHub project secret. If a maintainer could help me with that I'd be all for it.
 
As it stands, the value of the script is just as an enabler for contributors like myself, and if it does drift over time, then just like the manual testing docs, the expectation is that another contributor should help fix it up.

File: .github/infrastructure/conformance/conf-test-azure-cosmosdb.bicep [hunk 1]
```diff
@@ -0,0 +1,53 @@
+// ------------------------------------------------------------
```
[the remark this answers is not in the document]
Comment (CodeMonkeyLeet) [reply .github/infrastructure/conformance/conf-test-azure-cosmosdb.bicep]: Moved under `azure` subfolder as suggested

**Comment by artursouza** [comment]
Awesome. Can you add a README.md in how to use this for now?

Then, create an issue to create a workflow to setup the conformance test resources. Then, any new test resources would need to be created via the workflow and not manually on Azure Portal: infra as code :)

**Review by artursouza — APPROVED** [review]

**Pull request merged**